.github/workflows/rust.yml

```
@@ -1,39 +1,48 @@
-name: Rust
+name: Rust Build and Artifact Creation
 
 on:
   push:
-    branches: [ "master" ]
+    branches:
+      - main
+      - win-build
   pull_request:
-    branches: [ "master" ]
-
-env:
-  CARGO_TERM_COLOR: always
-  RUSTFLAGS: "-Dwarnings"
+    branches:
+      - main
+      - win-build
 
 jobs:
   build:
     strategy:
       matrix:
-        toolchain:
-          - channel: stable
-            cargo_args: --verbose --all-targets
-            clippy_args: ""
-          - channel: nightly
-            cargo_args: --verbose --all-targets --all-features
-            clippy_args: --allow clippy::needless_lifetimes
-
-    runs-on: ubuntu-latest
+        os: [ubuntu-latest, windows-latest]
+        toolchain: [stable, nightly]
+        include:
+          - os: ubuntu-latest
+            target: x86_64-unknown-linux-gnu
+          - os: windows-latest
+            target: x86_64-pc-windows-msvc
+    runs-on: ${{ matrix.os }}
 
     steps:
-    - uses: actions/checkout@v3
-    - name: Setup toolchain
-      run: >
-        rustup update ${{ matrix.toolchain.channel }} &&
-        rustup default ${{ matrix.toolchain.channel }} &&
-        rustup component add clippy
-    - name: Clippy
-      run: cargo clippy ${{ matrix.toolchain.cargo_args }} -- ${{matrix.toolchain.clippy_args}}
-    - name: Build
-      run: cargo build ${{ matrix.toolchain.cargo_args }}
-    - name: Run tests
-      run: cargo test ${{ matrix.toolchain.cargo_args }} -- --test-threads=1
+    - name: Checkout code
+      uses: actions/checkout@v3
+
+    - name: Set up Rust
+      uses: actions-rust-lang/setup-rust-toolchain@v1
+      with:
+        target: ${{ matrix.target }}
+        toolchain: ${{ matrix.toolchain }}
+        override: true
+
+    - name: Build and Install Binary
+      run: |
+        mkdir -p install_dir
+        cargo install --path . --root install_dir --target ${{ matrix.target }}
```

If you mean for the path to be the git root dir, use:
```suggestion
        cargo install --path  {{github.workspace}} --root install_dir --target ${{ matrix.target }}
```
Also, ideally don't use any relative paths whatsoever (so, --root should have an absolute path and mdir also, you can base all of them off f {{github.workspace}}